src/util/check.h

```diff
@@ -56,4 +56,16 @@ T get_pure_r_value(T&& val)
 /** Identity function. Abort if the value compares equal to zero */
 #define Assert(val) [&]() -> decltype(get_pure_r_value(val))& { auto& check = (val); assert(#val && check); return check; }()
 
+/** Helper for static_check_equal, no type inference **/
+template <typename T1, typename T2, T1 value_1, T2 value_2>
+struct static_check_equal_struct {
+    static_check_equal_struct() {
+        static_assert(value_1 == value_2, "Equality check failed, value_1 != value_2.");
+    }
+};
+
+/** Check if the template arguments are equal at compile time. Compiler error prints values if
+ * failure. **/
+#define static_check_equal(s1,s2) static_check_equal_struct<decltype(s1),decltype(s2),s1,s2>()
```

Comment:
clang-format suggestion
```suggestion
#define static_check_equal(s1, s2) static_check_equal_struct<decltype(s1), decltype(s2), s1, s2>()
```